### ISSUE TITLE
Iteration Plan for June 2020

### ISSUE THREAD

**Issue by kieferrm** [microsoft/vscode#100100]
This plan captures our work in **June**. We will ship in **early July**. This is a 4-week iteration.

## Endgame
- **June 29th, 2020**: Endgame begins
- **July 3rd, 2020**: Endgame done

The endgame details for this iteration are tracked in https://github.com/microsoft/vscode/issues/101289.

## Plan Items
Below is a summary of the top level plan items.

Legend of annotations:

| Mark                | Description                                       |
| ------------------- | ------------------------------------------------- |
| :runner:            | work in progress                                  |
| :hand:              | blocked task                                      |
| :muscle:            | stretch goal for this iteration                   |
| :red_circle:        | missing issue reference                           |
| :large_blue_circle: | more investigation required to remove uncertainty |
| :black_circle:      | under discussion within the team                  |
| :black_large_square:| a large work item, larger than one iteration      |

### Accessibility
- [x] Accessibility issues, see [query](https://github.com/Microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+label%3Aaccessibility+milestone%3A%22June+2020%22) @isidorn
- [x] Community call https://github.com/microsoft/vscode/issues/99916 @sana-ajani @isidorn

### Workbench
- [x] Review settings-sync end-to-end flow https://github.com/microsoft/vscode/issues/100426 @misolori @sandy081
- [ ] :runner: Use encrypted keychain for account management https://github.com/microsoft/vscode/issues/95475 @rmacfarlane @sbatten
- [x] Enable rich hovers across the workbench (custom trees, terminal) https://github.com/microsoft/vscode/issues/97496 @tyriar
- [x] Improve first run experience (open editors, npm scripts, ...) https://github.com/microsoft/vscode/issues/98614 @misolori @isidorn
- [x] Provide object editing support in the settings editor https://github.com/microsoft/vscode/issues/99635 @9at8
- [x] Improve Editor Inputs to better support moving and splitting of editors https://github.com/microsoft/vscode/issues/99770 @jrieken @rebornix @bpasero
- [ ] :runner: Undo/redo of workspace edit file operations, https://github.com/microsoft/vscode/issues/91383 @jrieken
- [x] Support multiple files in working copy file service https://github.com/microsoft/vscode/issues/98309 @bpasero
- [x] Git: Restore diff editors on reload https://github.com/microsoft/vscode/issues/99290 @bpasero

### Notebook Editor
- [x] Support for Python adoption https://github.com/microsoft/vscode-python/issues/10496#issuecomment-626028704 @rebornix
- [x] Stabilization [query](https://github.com/microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+label%3Anotebook+label%3Abug) @roblourens @rebornix
- [x] Polish notebook behavior [query](https://github.com/microsoft/vscode/issues?q=is%3Aissue+is%3Aopen+label%3Anotebook+milestone%3A%22June+2020%22+label%3Apolish) @roblourens @rebornix
- [x] Improve notebook UX using indentation, borders, colors [query](https://github.com/microsoft/vscode/issues?q=is%3Aissue+is%3Aopen+label%3Anotebook+milestone%3A%22June+2020%22+label%3Aux) @misolori
- [x] Provide a yeoman generator for the notebook renderer starter https://github.com/microsoft/vscode/issues/101056 @connor4312
- [x] Develop and end-to-end sample that maps many cells to a single LS document, https://github.com/microsoft/vscode/issues/100186 @jrieken
- [x] Provide a notebook SDK (documentation, samples) https://github.com/microsoft/vscode-docs/issues/3784 @jacksonkearl
- [x] Explore a PowerBI custom output renderer https://github.com/microsoft/vscode/issues/100759 @sbatten
- [x] Allow to edit markdown documents as notebooks for better self-hosting coverage https://github.com/microsoft/vscode/issues/100307 @roblourens
- [x] Investigate how to make persisted cell breakpoints robust https://github.com/microsoft/vscode/issues/99943 @weinand
- [x] Finish the notebook debug sample extension including support for simplified cell debugging https://github.com/microsoft/vscode/issues/99945 https://github.com/microsoft/vscode/issues/99946 @weinand

#### LSP / LSIF
- [x] [Finish support for semantic coloring in the LSP spec](https://github.com/microsoft/language-server-protocol/issues/18) @dbaeumer
- [x] Support to defer the calculation of `additionalTextEdits` until the completion item is resolved https://github.com/Microsoft/language-server-protocol/issues/1003 @dbaeumer

#### TypeScript/JavaScript
- [X] Use TS v4.0 for our building VS Code https://github.com/microsoft/vscode/issues/100796 @mjbvz

### Debug
- [x] Improve the single file debug experience https://github.com/microsoft/vscode/issues/92269 @weinand @isidorn
- [x] Explore how to make the debug terminal easier to discover https://github.com/microsoft/vscode/issues/98054 @connor4312
- [x] Investigate support of `webview` debugging https://github.com/microsoft/vscode/issues/96626 @connor4312 @weinand
- [x] Unclutter the Call Stack view by eliminating parent debug session https://github.com/microsoft/vscode/issues/99736 @isidorn @connor4312 @weinand

### SCM
- [x] Unify SCM views to support the flexible workbench layout https://github.com/microsoft/vscode/issues/90461 @joaomoreno

### API
- API finalization: [query](https://github.com/Microsoft/vscode/issues?q=is%3Aissue+is%3Aopen+milestone%3A%22June+2020%22+label%3A%22api-finalization%22)
- API proposals: [query](https://github.com/Microsoft/vscode/issues?q=is%3Aissue+is%3Aopen+milestone%3A%22June+2020%22+label%3A%22api-proposal%22)

### Web
- [x] Improve the support for large file uploads https://github.com/microsoft/vscode/issues/83565 @bpasero

### Extensions
- [x] Review client/marketplace interactions https://github.com/microsoft/vscode/issues/98366 @sandy081
- [x] Update to the latest extension category names https://github.com/microsoft/vscode/issues/100399 @sandy081 @fiveisprime
- [x] Support badges from GitHub actions https://github.com/microsoft/vscode-vsce/issues/389 depends on https://github.com/microsoft/vsmarketplace/issues/19 @joaomoreno

### Extension Contributions
- [x] Improvements to the Remote extensions https://github.com/microsoft/vscode-remote-release/issues/3175 **team**
- [x] Improvements to the GitHub PR and Issues extension, see [query](https://github.com/microsoft/vscode-pull-request-github/issues?q=is%3Aopen+is%3Aissue+assignee%3Aalexr00+milestone%3A%22June+2020%22) @alexr00
- [x] Improvements to the Hex editor extension https://github.com/microsoft/vscode-hexeditor/issues/48 @lramos15
- [x] Investigate SDK deprecations for the Azure Account extensions https://github.com/microsoft/vscode-azure-account/issues/140 @rmacfarlane

### Install
- [ ] :runner: Explore `msix` installer for VS Code on Windows https://github.com/microsoft/vscode/issues/61793 @sbatten
- [x] Explore `winget` support (update manifest on release, support insiders version) https://github.com/microsoft/vscode/issues/100745 @lszomoru @fiveisprime

### Engineering
- [x] Provide `ripgrep` binaries for Windows on ARM https://github.com/microsoft/vscode/issues/98587 @roblourens

### Bots
- [X] Improve issue bot models by training them on GPUs https://github.com/microsoft/vscode/issues/100202 @jacksonkearl
- [X] Publish code of the issue triaging extension https://github.com/microsoft/vscode-github-triage-extension @jacksonkearl
- [ ] :hand: Make issue triaging extension available in browser store(s) https://github.com/microsoft/vscode-github-triage-extension/issues/2 @jacksonkearl

### Electron
- [x] Investigate start-up performance issues when using electron 8 https://github.com/microsoft/vscode/issues/97638 @deepak1556 @jrieken
- [ ] :runner: Upgrade to electron 8 https://github.com/microsoft/vscode/issues/100731 @deepak1556

### Node.js-free, sandboxed renderer :black_large_square:
Backlog: https://github.com/microsoft/vscode/issues/92164
- [X] Debt: adopt service based IPC communication over raw IPC @bpasero
- [X] OOPIF support with custom protocols for desktop https://github.com/microsoft/vscode/issues/95955 @mjbvz @deepak1556 @alexdima
- [ ] :runner: Remove dependency on node for startup path https://github.com/microsoft/vscode/issues/98682 :black_large_square: @deepak1556 @bpasero
   - [ ] adopt V8 cached data option for browser windows @deepak1556
   - [ ] move off `file://` protocol for serving static assets @deepak1556
- [ ] :runner: Support for encodings in sandboxed/web environments (`iconv-lite`) https://github.com/microsoft/vscode/issues/79275 @bpasero
- [ ] Minimal sandboxed version of the workbench with stubbed services :black_large_square: @bpasero @deepak1556
- [ ] :black_large_square: enable `sandbox: true` and `contextIsolation` for some of our windows to get our feet wet @deepak1556
  - [x] proxy auth dialog
  - [ ] issue reporter
  - [ ] process explorer

### Documentation
- None.

-----
### Deferred Items
- Improve existing Windows installer https://github.com/microsoft/vscode/issues/61793 @lszomoru @joaomoreno
- :red_circle: Offer to create fork on GitHub when user has no commit rights to a repository @joaomoreno
- Simplify the Git '...' menu and introduce submenus, depends on https://github.com/microsoft/vscode/issues/100172 @joaomoreno
- First run experience: simplify Extensions view overflow menu https://github.com/microsoft/vscode/issues/100756 @sandy081
- Investigate into contentEditable for the hidden accessibility 'textArea' https://github.com/microsoft/vscode/issues/97154 @isidorn
- Adding a table widget (used for example in a variable/data view in the debugger) https://github.com/microsoft/vscode/issues/99997 @isidorn
- Investigate issues introduced by the Web UI stopped indication https://github.com/microsoft/vscode/issues/100757 @weinand @isidorn
- :muscle: Explore tabular variables debug view (depends on adding a table widget see above) https://github.com/microsoft/vscode/issues/100205 @weinand @isidorn
- :muscle: Support Jupyter-like cell folding https://github.com/microsoft/vscode/issues/100308 @roblourens
- :muscle: Support Linux ARM64 https://github.com/microsoft/vscode/issues/6442 @joaomoreno
- :muscle: Provide guidelines for extension authors how to adopt the GH auth provider https://github.com/microsoft/vscode/issues/100995 @rmacfarlane

**Comment by gjsjohnmurray** [comment]
Based on https://github.com/microsoft/vscode/issues/93598#issuecomment-640893897 it looks like there's a better issue to link object editing support to.

**Comment by yubaoquan** [comment]
Why not skip electron 8, use 9 instead?
I remember there is a history vscode skipped a major version of electron.

**Comment by deepak1556** [comment]
@yubaoquan we found out quite a few blocking issues when we started exploring Electron 8 https://github.com/microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+label%3Aelectron-8-update, jumping versions would only make it difficult to narrow down the issue and get them fixed in a timely manner. When we push Electron 8 to insiders, Electron 9 will adopted into our exploration builds, upgrading to Electron 9 was also blocked by https://github.com/microsoft/vscode/issues/95955 which got fixed yesterday.

To top it all I would say, its too soon to skip versions :)
